include/oneapi/dpl/pstl/algorithm_ranges_impl.h

```diff
@@ -144,6 +144,71 @@ __pattern_find_if(__serial_tag</*IsVector*/std::false_type>, _ExecutionPolicy&&,
     return std::ranges::find_if(std::forward<_R>(__r), __pred, __proj);
 }
 
+//---------------------------------------------------------------------------------------------------------------------
+// __pattern_find_first_of
+//---------------------------------------------------------------------------------------------------------------------
+
+template <typename _Tag, typename _ExecutionPolicy, typename _R1, typename _R2, typename _Pred, typename _Proj1, typename _Proj2>
+auto
+__pattern_find_first_of(_Tag __tag, _ExecutionPolicy&& __exec, _R1&& __r1, _R2&& __r2,
+                        _Pred __pred, _Proj1 __proj1, _Proj2 __proj2)
+{
+    static_assert(__is_parallel_tag_v<_Tag> || typename _Tag::__is_vector{});
+
+    auto __bin_pred = [__pred, __proj1, __proj2](auto&& __val1, auto&& __val2) {
+        return std::invoke(__pred, std::invoke(__proj1, std::forward<decltype(__val1)>(__val1)),
+                           std::invoke(__proj2, std::forward<decltype(__val2)>(__val2)));
```

Comment:
Lambdas have been replaced in https://github.com/uxlfoundation/oneDPL/pull/2153. Let's avoid adding new ones.

Reply:
I've found 10 lambdas at the device code path (sorry, `algorithm_ranges_impl.h` was a bad file to point out this issue). Hence, these 10 lambdas should be replaced with functors, which is repetitive, but not overwhelming.

I would suggest fixing it here if the time allows, or in a follow-up PR.

Reply:
Actually we have only 3 days. I am on vacation the next week.
In any case, formal changing the all lambda in new range based API code is not a subject of this PR.

Again, 
A possible "Lambdas"  issue (in some tricky cases and with unnamed lambda OFF) was detected AFTER this PR 2100 was done. It is not a subject of merging this PS and the main branch. The change lambdas by functors - it is the new code and should be considered as a new separate PR. It is my opinion.